src/sonic-bgpcfgd/tests/test_static_rt.py

```diff
@@ -0,0 +1,420 @@
+            "nexthop-vrf": "nh_vrf,,default",
+            "blackhole": "false,false,false",
+        }),
+        True,
+        [
+            "ip route 10.1.3.0/24 10.0.0.57 PortChannel0001 10 nexthop-vrf nh_vrf vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.59 PortChannel0002 20 vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.61 PortChannel0003 30 nexthop-vrf default vrf vrfRED"
+        ]
+    )
+
+def test_set_same_route():
+    mgr = constructor()
+    set_del_test(
+        mgr,
+        "SET",
+        ("vrfRED|10.1.3.0/24", {
+            "nexthop": "10.0.0.57,10.0.0.59,10.0.0.61",
+            "ifname": "PortChannel0001,PortChannel0002,PortChannel0003",
+            "distance": "10,20,30",
+            "nexthop-vrf": "nh_vrf,,default",
+            "blackhole": "false,false,false",
+        }),
+        True,
+        [
+            "ip route 10.1.3.0/24 10.0.0.57 PortChannel0001 10 nexthop-vrf nh_vrf vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.59 PortChannel0002 20 vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.61 PortChannel0003 30 nexthop-vrf default vrf vrfRED"
+        ]
+    )
+    set_del_test(
+        mgr,
+        "SET",
+        ("vrfRED|10.1.3.0/24", {
+            "nexthop": "10.0.0.57,10.0.0.59,10.0.0.61",
+            "ifname": "PortChannel0001,PortChannel0002,PortChannel0003",
+            "distance": "40,50,60",
+            "nexthop-vrf": "nh_vrf,,default",
+            "blackhole": "false,false,false",
+        }),
+        True,
+        [
+            "no ip route 10.1.3.0/24 10.0.0.57 PortChannel0001 10 nexthop-vrf nh_vrf vrf vrfRED",
+            "no ip route 10.1.3.0/24 10.0.0.59 PortChannel0002 20 vrf vrfRED",
+            "no ip route 10.1.3.0/24 10.0.0.61 PortChannel0003 30 nexthop-vrf default vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.57 PortChannel0001 40 nexthop-vrf nh_vrf vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.59 PortChannel0002 50 vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.61 PortChannel0003 60 nexthop-vrf default vrf vrfRED"
+        ]
+    )
+
+def test_set_add_nh():
+    mgr = constructor()
+    set_del_test(
+        mgr,
+        "SET",
+        ("vrfRED|10.1.3.0/24", {
+            "nexthop": "10.0.0.57,10.0.0.59,10.0.0.61",
+            "ifname": "PortChannel0001,PortChannel0002,PortChannel0003",
+            "distance": "10,20,30",
+            "nexthop-vrf": "nh_vrf,,default",
+            "blackhole": "false,false,false",
+        }),
+        True,
+        [
+            "ip route 10.1.3.0/24 10.0.0.57 PortChannel0001 10 nexthop-vrf nh_vrf vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.59 PortChannel0002 20 vrf vrfRED",
+            "ip route 10.1.3.0/24 10.0.0.61 PortChannel0003 30 nexthop-vrf default vrf vrfRED"
+        ]
+    )
+    set_del_test(
+        mgr,
+        "SET",
+        ("vrfRED|10.1.3.0/24", {
+            "nexthop": "10.0.0.57,10.0.0.59,10.0.0.61,10.0.0.63",
```

Comment:
please also have a test to remove one nexthop

Reply:
Added the scenario of removing part of the nexthops.